### PR TITLE
[tests] new env var TAR_SCM_TC to select test cases

### LINKED ISSUE #146: test suite needs optimising
It's currently too hard to get quick feedback from the test suite, as there are too many tests taking too long.  We should do some performance profiling on the tests and tackle the bottlenecks.

### PR DESCRIPTION
**Pull request by M0ses**
Without this patch you have to edit the tests/test.py and comment out
tests you dont want to run in an array called testclasses.

This patch gives you the possibilty to run only selected testclasses
by specifing an environment variable with a comma separated list of
testclasses to run, e.g.

TAR_SCM_TC=UnitTestCases,TasksTestCases PYTHONPATH=. tests/test.py

This is only the first step in direction to fix #146 and #147

### CODE REVIEW

**Comment by M0ses** [comment]
In this commit I splitted up each single pylint fix into a commit, but this is very inefficient as I`m writing more commit messages than code. Could we agree to do all the lint fixes in one commit (including the pylint report in the commit message)?

**Comment by aspiers** [comment]
I would prefer to keep them separate otherwise there are way too many different types of change to review.  However I think this is one of the rare occasions where a detailed commit message is not needed - as long as the commit message title makes it clear which category of pylint check is being fixed, that should be sufficient.  So the overhead should not be big.  A way to reduce it even further is to follow your good suggestion of submitting a commit per category of pylint check which fixes issues across the whole codebase for that category.

**Comment by M0ses** [comment]
Ok, what do you think about the change despite the lint changes? I hate to edit the tests/test.py to get rid of the other tests, especial because bzr based test cases are broken permanently as bzr is broken in TW ATM

**Comment by aspiers** [comment]
I'm in Israel but back Wed. Should hopefully get time to review this in next day or 3.

File: tests/test.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def str_to_class(string):
     return getattr(sys.modules[__name__], string)
 
 if __name__ == '__main__':
-    suite = unittest.TestSuite()
+    SUITE = unittest.TestSuite()
```
Comment (M0ses) [inline tests/test.py]: Even if i isolate the into two functions (prepare_testclasses and prepare_testsuite) the problem with the constants persists. I don`t care if we disable the pylint checks as suggested in the link you posted (solution 2) or if we simply write the variable name in upper case letters.

the problem persists for

* suite 
* runner
* result

**Pull request merged**